### PR TITLE
Prepare release 5.1.0

### PR DESCRIPTION
**Pull request by Premwoik**
This PR includes final changes before release 5.1.0

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125060) / small_tests / f4d67e9e00fc63e284a92ae2af97731e31c76199
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125060/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125060/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2022-06-02_07.01.46/lib.mongooseim.logs/run.2022-06-02_07.01.46/suite.log.html)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,250 @@
...
+- Update CONTRIBUTING.md (#3622)
+
+## Commits, merged PRs and closed issues
+- [List of merged PRs](https://github.com/esl/MongooseIM/pulls?q=is%3Apr+is%3Amerged+milestone%3A5.1.0)
+
+- [List of closed issues](https://github.com/esl/MongooseIM/issues?q=is%3Aissue+is%3Aclosed+closed%3A2021-10-08..2022-06-09)
+
+- [Repository history for this release](https://github.com/esl/MongooseIM/graphs/contributors?from=2021-10-08&to=2022-06-09&type=c)
+
+[List of merged PRs based on merge date]: # (https://github.com/esl/MongooseIM/pulls?q=is%3Apr+base%3Amaster+merged%3A%222021-10-08..2022-06-09%22+sort%3Acreated-asc+)
```
Comment (chrzaszcz) [inline CHANGELOG.md:243]: This looks suspicious - it is not rendered in the Markdown view.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,250 @@
...
+   - Remove all calls to `gen_mod:set_module_opt` from modules, making the module config constant(#3389)
+   - Remove calls to `gen_mod:set_module_opt(s)` in big tests (#3398)
+   - Remove `gen_mod:set_module_opt(s)` (#3468)
+   - Remove the ETS table from `gen_mod` (#3484)
+   - Eliminate `gen_mod:get_opt_subhost` (#3626)
+- Remove dynamically compiled modules (#3394)
+   - POC module `mod_private` (#3323)
+   - Remove `ejabberd_rdbms` and `mongoose_rdbms_type` (#3392)
+   - Allow global to be passed into `mongoose_backend` as an arg (#3386)
+   - MAM cassandra backend without dynamicly compiled modules (#3373)
```
Comment (chrzaszcz) [inline CHANGELOG.md:40]:
```suggestion
   - MAM cassandra backend without dynamically compiled modules (#3373)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,250 @@
...
+- Removed in internal configuration rework
+   - Remove all calls to `gen_mod:set_module_opt` from modules, making the module config constant(#3389)
+   - Remove calls to `gen_mod:set_module_opt(s)` in big tests (#3398)
+   - Remove `gen_mod:set_module_opt(s)` (#3468)
+   - Remove the ETS table from `gen_mod` (#3484)
+   - Eliminate `gen_mod:get_opt_subhost` (#3626)
+- Remove dynamically compiled modules (#3394)
+   - POC module `mod_private` (#3323)
+   - Remove `ejabberd_rdbms` and `mongoose_rdbms_type` (#3392)
+   - Allow global to be passed into `mongoose_backend` as an arg (#3386)
```
Comment (chrzaszcz) [inline CHANGELOG.md:39]:
```suggestion
   - Allow `global` to be passed into `mongoose_backend` as an arg (#3386)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,250 @@
...
+   - Use `mongoose_acc` timestamp in MAM (#3320)
+   - Same stanza id for peers (#3376)
+   - Retract on stanza ids (#3377, #3384, #3385)
+   - Retraction events (#3497, #3498, #3513)
+   - Report wrong stanza id more friendly (#3591)
+   - Make MAM return timestamps in microseconds (#3595)
+   - Reorder MAM's `filter_local_packet` hook order (#3623)
+   - MAM shaper improvements (#3641)
+- Smart markers improvements (#3590)
+   - Implement all the automatic removals (e. g. `remove_domain`) for the smart markers (#3544)
```
Comment (chrzaszcz) [inline CHANGELOG.md:101]:
```suggestion
   - Implement all the automatic removals (e.g. `remove_domain`) for the smart markers (#3544)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,250 @@
...
+   - Prepare a documentation page for smart markers (#3535)
+   - Implement all configurations for the smart markers (#3546, #3592)
+- Inbox improvements (#3359, #3596, #3604)
+   - Improve types for dialyzer and log errors when they happen (#3366)
+   - Simplify time-unit conversions (#3405)
+   - Fix inbox filter bug that puts unnecessary load on the auth backend (#3449)
+   - Inbox refactor tests (#3452)
```
Comment (chrzaszcz) [inline CHANGELOG.md:108]:
```suggestion
   - Refactor Inbox tests (#3452)
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,250 @@
...
+- Less MIM compiler warnings (#3524)
+- Add config flag to be able to set sni hostname matching (#3528)
+- Fix `is_configured` in async pools (#3566)
+- Do not call the obsolete backend API in CT hook (#3570)
+- Fix metrics for auth backend not being reported (#3585)
+- Fix `cowboy_static` options (#3618)
+- Fix `mod_ping` with `host_types` (#3638)
+- Use metric probes instead of eval for slow calculated metrics (#3644)
+- Fix flaky tests and speed up CI:
+   - Gather list of failed test cases in separate issue (#3361)
```
Comment (chrzaszcz) [inline CHANGELOG.md:207]:
```suggestion
   - Gather list of failed test cases in a separate issue (#3361)
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,250 @@
...
+   - Fix `rest_client_SUITE:muc:messages_can_be_paginated_in_room` (#3363)
+   - Fill the gaps in the event table for `service_domain_db` (#3365)
+   - Run muclight inbox tests one by one instead of parallel (#3374)
+   - Print preset in tools/circle-publish-github-summary-comment.sh (#3378)
```
Comment (chrzaszcz) [inline CHANGELOG.md:211]:
```suggestion
   - Print preset in `tools/circle-publish-github-summary-comment.sh` (#3378)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,250 @@
...
+   - Reroute presences correctly on reconnect (SM fix) (#3459)
+   - Wait for async writers to flush messages in tests (#3466)
+   - Inbox flaky tests fixes (#3630)
+
+### Other
+
+- Faster, more stable CircleCI on docker with DB backends (#3401, #3519, #3530)
+- Updating dependencies and cleaning the rebar.lock file (#3425, #3430, #3465, #3490, #3520, #3531, #3533, #3573, #3657)
+- Document low-level MAM options (#3329)
+- Document  MUC Light cache affiliations (#3509)
```
Comment (chrzaszcz) [inline CHANGELOG.md:232]:
```suggestion
- Document MUC Light cache affiliations (#3509)
```

**Review by chrzaszcz — COMMENTED** [review]
It looks good in general, I added a few comments to the changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (Premwoik) [reply CHANGELOG.md:243]: I think it is a comment or smth like this. It was like this in the previous releases, and I copied it.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125323) / small_tests / 40113966be92f675984142cd61c8a3cba7e85631
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125323/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125323/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-03_06.34.50/lib.mongooseim.logs/run.2022-06-03_06.34.51/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125353) / small_tests / 03c346e5ce74fddd733ece2dcd744fe8a89c2b19
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125353/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125353/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-03_08.43.52/lib.mongooseim.logs/run.2022-06-03_08.43.52/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125610) / small_tests / 84d183ddd0466189b66325464b8e5308a783bfdb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125610/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125610/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-06_08.32.13/lib.mongooseim.logs/run.2022-06-06_08.32.14/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125925) / small_tests / 6218a46c9d4084f31f376c359fc08c0a78ac7e16
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125925/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125925/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-07_14.17.06/lib.mongooseim.logs/run.2022-06-07_14.17.07/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/125956) / small_tests / 2186449d86d735cee0d74c31d11f47eadb950651
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/125956/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/125956/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-08_07.24.04/lib.mongooseim.logs/run.2022-06-08_07.24.04/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/126005) / small_tests / d8d8cc37996ff1280ef3a1a9824b6482b59dbca9
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3666/126005/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3666/126005/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-06-08_07.57.55/lib.mongooseim.logs/run.2022-06-08_07.57.56/suite.log.html)

**Pull request merged**